unsafe/src/main/java/org/apache/spark/unsafe/types/UTF8String.java

```diff
@@ -178,6 +189,7 @@ public UTF8String clone() {
 
   @Override
   public int compareTo(final UTF8String other) {
+    if (other == null) return 1;
```

Comment:
The expression `RowOrdering` actually did the null testing already before call the `compareTo`, see https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/rows.scala#L218
So, we are sure the argument passed in will be non-null.